Modix.Services/Core/MessageLogBehavior.cs

```diff
@@ -146,12 +153,52 @@ private async Task HandleMessageDelete(Cacheable<IMessage, ulong> message, ISock
                 .WithDescription(descriptionContent)
                 .WithCurrentTimestamp();
 
+            await SelfExecuteRequest<IMessageRepository>(async messages => await messages.RemoveAsync(message.Id));
+
             await SelfExecuteRequest<IDesignatedChannelService>(async designatedChannelService =>
             {
                 await designatedChannelService.SendToDesignatedChannelsAsync(
                     guild, DesignatedChannelType.MessageLog,
                     $":wastebasket:Message Deleted in {MentionUtils.MentionChannel(channel.Id)} `{message.Id}`", embed.Build());
             });
         }
+
+        private async Task HandleMessageReceived(SocketMessage message)
+        {
+            Log.LogDebug("Handling message received event for message #{MessageId}.", message.Id);
+
+            if (!message.Content.StartsWith('!') &&
+                message.Channel is IGuildChannel channel &&
+                message.Author is IGuildUser author &&
+                author.Guild is SocketGuild guild &&
+                !author.IsBot && !author.IsWebhook)
+            {
+                await SelfExecuteRequest<IMessageRepository>(
+                    async messages =>
+                    {
+                        Log.LogInformation("Logging message #{MessageId} to the database.", message.Id);
+
+                        var entity = new MessageEntity
+                        {
+                            Id = message.Id,
+                            GuildId = guild.Id,
+                            ChannelId = channel.Id,
+                            AuthorId = author.Id,
+                            Timestamp = message.Timestamp
+                        };
+
+                        Log.LogDebug("Entity for message #{MessageId}: {@Message}", message.Id, entity);
+
+                        try
```

Comment:
`.SelfExecuteRequest` already has a catch-all exception handler. Unless you can actually handle specific errors and recover from them, you should just let the exception bubble. If you want to just provide some meaningful information about the error, wrap the exception and re-throw it.

Reply:
One man's abundance of caution is another man's clutter.

Reply:
This can/will probably be refactored later on once we have better service-level error handling and custom exceptions. Still fine as-is since it'll still be logged properly if something goes wrong.